vector/src/test/java/im/vector/app/features/settings/devices/v2/othersessions/OtherSessionsViewModelTest.kt

```diff
@@ -0,0 +1,292 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.settings.devices.v2.othersessions
+
+import android.os.SystemClock
+import com.airbnb.mvrx.Success
+import com.airbnb.mvrx.test.MavericksTestRule
+import im.vector.app.features.settings.devices.v2.DeviceFullInfo
+import im.vector.app.features.settings.devices.v2.GetDeviceFullInfoListUseCase
+import im.vector.app.features.settings.devices.v2.RefreshDevicesUseCase
+import im.vector.app.features.settings.devices.v2.details.extended.DeviceExtendedInfo
+import im.vector.app.features.settings.devices.v2.filter.DeviceManagerFilterType
+import im.vector.app.features.settings.devices.v2.list.DeviceType
+import im.vector.app.test.fakes.FakeActiveSessionHolder
+import im.vector.app.test.fakes.FakeVerificationService
+import im.vector.app.test.test
+import im.vector.app.test.testDispatcher
+import io.mockk.every
+import io.mockk.mockk
+import io.mockk.mockkStatic
+import io.mockk.unmockkAll
+import io.mockk.verifyAll
+import kotlinx.coroutines.flow.flowOf
+import org.junit.After
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.matrix.android.sdk.api.session.crypto.model.DeviceInfo
+import org.matrix.android.sdk.api.session.crypto.model.RoomEncryptionTrustLevel
+
+private const val A_TITLE_RES_ID = 1
+private const val A_DEVICE_ID = "device-id"
+
+class OtherSessionsViewModelTest {
+
+    @get:Rule
+    val mavericksTestRule = MavericksTestRule(testDispatcher = testDispatcher)
+
+    private val defaultArgs = OtherSessionsArgs(
+            titleResourceId = A_TITLE_RES_ID,
+            defaultFilter = DeviceManagerFilterType.ALL_SESSIONS,
+            excludeCurrentDevice = false,
+    )
+
+    private val fakeActiveSessionHolder = FakeActiveSessionHolder()
+    private val fakeGetDeviceFullInfoListUseCase = mockk<GetDeviceFullInfoListUseCase>()
+    private val fakeRefreshDevicesUseCaseUseCase = mockk<RefreshDevicesUseCase>()
+
+    private fun createViewModel(args: OtherSessionsArgs = defaultArgs) = OtherSessionsViewModel(
+            initialState = OtherSessionsViewState(args),
+            activeSessionHolder = fakeActiveSessionHolder.instance,
+            getDeviceFullInfoListUseCase = fakeGetDeviceFullInfoListUseCase,
+            refreshDevicesUseCase = fakeRefreshDevicesUseCaseUseCase,
+    )
+
+    @Before
+    fun setup() {
+        // Needed for internal usage of Flow<T>.throttleFirst() inside the ViewModel
+        mockkStatic(SystemClock::class)
+        every { SystemClock.elapsedRealtime() } returns 1234
+
+        givenVerificationService()
```

Comment:
The returned value is not used, so maybe make `givenVerificationService()` returns `Unit`

Reply:
In fact, I forgot to add some unit tests about adding and removing listener on the verification service. It will be updated in a following PR.